spec/locales-currencies-tz.html

```diff
@@ -224,10 +319,29 @@ <h1>AvailableNamedTimeZoneIdentifiers ( ): a List of Time Zone Identifier Record
         1. Let _result_ be a new empty List.
         1. For each element _identifier_ of _identifiers_, do
           1. Let _primary_ be _identifier_.
-          1. If _identifier_ is a Link name and _identifier_ is not *"UTC"*, then
-            1. Set _primary_ to the Zone name that _identifier_ resolves to, according to the rules for resolving Link names in the IANA Time Zone Database.
-            1. NOTE: An implementation may need to resolve _identifier_ iteratively.
-          1. If _primary_ is one of *"Etc/UTC"*, *"Etc/GMT"*, or *"GMT"*, set _primary_ to *"UTC"*.
+          1. If _identifier_ is listed in the first column of <emu-xref href="#table-legacy-time-zone-identifier-mapping"></emu-xref>, then
+            1. Set _primary_ to the value in the second column of the row where the first column is _identifier_.
+          1. Else,
+            1. NOTE: The algorithm steps below are intended to mimic the behaviour of <code>icu::TimeZone::getIanaID()</code> in the International Components for Unicode (<a href="https://icu.unicode.org/">ICU</a>).
+               The steps in this section are provided for testing conformance of ICU-based ECMAScript implementations, and to ensure compatibility between ICU-based and non-ICU-based implementations.
+            1. If _identifier_ is present in the "TZ" column of *zone.tab* of the IANA Time Zone Database, then
+              1. Set _primary_ to _identifier_.
+            1. Else if _identifier_ is a Link name in the IANA Time Zone Database, then
+              1. Let _zone_ be the Zone name that _identifier_ resolves to, according to the rules for resolving Link names in the IANA Time Zone Database.
+              1. If _zone_ starts with *"Etc/"*, then
+                1. Set _primary_ to _zone_.
+              1. Else,
+                1. Let _identifierCountryCode_ be the <a href="https://www.iso.org/glossary-for-iso-3166.html">ISO 3166-1 Alpha-2</a> country code whose territory contains the geographical area corresponding to _identifier_.
+                1. Let _zoneCountryCode_ be the ISO 3166-1 Alpha-2 country code whose territory contains the geographical area corresponding to _zone_.
+                1. If _identifierCountryCode_ is _zoneCountryCode_, then
+                  1. Set _primary_ to _zone_.
+                1. Else,
+                  1. Let _countryCodeLine_ be the line in file <code>zone.tab</code> of the IANA Time Zone Database where the "country-code" column is _identifierCountryCode_.
```

Comment:
As of right now, 31 codes have more than one line in zone.tab:

<details>

```console
$ curl -sSL 'https://github.com/eggert/tz/raw/main/zone.tab' \
  | awk '{ if (match($1, "^[A-Z][A-Z]$")) print $1; }' \
  | sort \
  | uniq -c \
  | awk '$1 > 1 { print; i++; } END { print i }'
     10 AQ
     12 AR
     12 AU
     16 BR
     23 CA
      2 CD
      3 CL
      2 CN
      2 CY
      2 DE
      2 EC
      3 ES
      3 FM
      4 GL
      4 ID
      3 KI
      7 KZ
      2 MH
      3 MN
     12 MX
      2 MY
      2 NZ
      3 PF
      2 PG
      2 PS
      3 PT
     26 RU
      2 UA
      2 UM
     29 US
      2 UZ
31
```

</details>

For example, America/Blanc-Sablon (CA) is a Link to America/Puerto_Rico (PR), but CA has a total of 23 lines in <tt>zone.tab</tt> (including one that is specifically for America/Blanc-Sablon), and similar arrangements affect regions as populous as Africa/Kinshasa and Asia/Kuala_Lumpur. So this algorithm is underspecified. What is the actual intent?

Reply:
Ahh, shoot. Yeah, we'll need a better solution, thanks for catching this. 

The problem to solve here is multi-hop Links like A => B => C where A and B are in the the same country but C is in a different country. This happens when an earlier TZDB release created the A => B Link, then a later release folded B into C, so the link that used to be A => B now was changed to A => C (along with a new Link for B => C).

There are 15 cases like this, conveniently separated into three cases of 5 each. 

**Cases where A is in zone.tab**
- America/Kralendijk ⇨ America/Curacao  ⇨ America/Puerto_Rico
- America/Lower_Princes ⇨ America/Curacao ⇨ America/Puerto_Rico
- America/Marigot ⇨ America/Port_of_Spain ⇨ America/Puerto_Rico
- America/St_Barthelemy ⇨ America/Port_of_Spain  ⇨ America/Puerto_Rico
- Arctic/Longyearbyen ⇨ Europe/Oslo ⇨ Europe/Berlin

**Cases where A's country code has only one line in zone.tab**
- America/Virgin ⇨ America/St_Thomas  ⇨ America/Puerto_Rico
- Atlantic/Jan_Mayen ⇨ Europe/Oslo ⇨ Europe/Berlin
- Iceland ⇨ Atlantic/Reykjavik ⇨ Africa/Abidjan
- Africa/Asmera ⇨ Africa/Asmara ⇨ Africa/Nairobi
- Africa/Timbuktu ⇨ Africa/Bamako ⇨ Africa/Abidjan

**Cases where A's country code has multiple lines in zone.tab (broken case in current PR's spec text)**
- (FM) Pacific/Ponape ⇨ Pacific/Pohnpei ⇨ Pacific/Guadalcanal
- (FM) Pacific/Truk ⇨ Pacific/Chuuk ⇨ Pacific/Port_Moresby
- (FM) Pacific/Yap ⇨ Pacific/Chuuk ⇨ Pacific/Port_Moresby
- (CA) America/Coral_Harbour ⇨ America/Atikokan ⇨ America/Panama
- (AQ) Antarctica/South_Pole ⇨ Antarctica/McMurdo ⇨ Pacific/Auckland 

For that last case, the only way to know what B was is to look in the `backzone` file in TZDB. The old Links are listed there. 

I'll draft some spec text to handle that case, and will let you know once I push it because I'm sure it will require a round or two of polishing. I suspect that the cleanest solution will be to just to use `backzone` for both the second and third cases above, and remove the complex spec text to handle the second case using zone.tab.

Thanks again for catching this.

Reply:
@gibson042 I just pushed a solution to the problem you found, as well as fixing (I think) all the other review feedback you had except the quote characters. Want to review the latest?

For the cases above, it turned out that there wasn't a single algorithmic solution possible, so I used the existing PR's algorithm for Case 2 above, and used `backzone` for Case 3. 

Sadly, we can't use `backzone` for both cases 2 and 3 because a single A => B => C chain in `backzone` requires the zone.tab algorithm. The problematic identifier is Atlantic/Jan_Mayen. This ID is linked to Europe/Oslo (country code "NO") in `backzone`, but it should be linked to Arctic/Longyearbyen which has the same country code ("SJ") as Atlantic/Jan_Mayen.

Let me know what you think of the latest commit.